### PR TITLE
Add docstrings for array module

### PR DESCRIPTION
**Pull request by MSanKeys963**
xref: #2215 and #2245 

This PR adds docstrings for `zarr.array` module and builds on top of @e-marshall's #2276. I've added documentation for following functions:

- from_dict()
- getitem()
- setitem()
- resize()
- update_attributes()

I will start adding code examples soon. Feedback on the current draft would be greatly appreciated. Thanks!

### CODE REVIEW

**Comment by MSanKeys963** [comment]
pre-commit.ci autofix

**Comment by jhamman** [comment]
@MSanKeys963 - we just merged @e-marshall's PR so you can update this one and finish it off. 

I think you can also skip docstrings for any method that start with `_`. So that should help limit the scope here.

**Comment by MSanKeys963** [comment]
Please review, @jhamman — my apologies for the delay.

File: src/zarr/core/array.py [hunk 1]
```diff
@@ -2798,6 +2949,30 @@ def resize(self, new_shape: ChunkCoords) -> Array:
         return type(self)(_resized)
 
     def update_attributes(self, new_attributes: dict[str, JSON]) -> Array:
+        """
+        Update the array's attributes.
+
+        Parameters
+        ----------
+        new_attributes : dict of str to JSON
```
Comment (jhamman) [inline src/zarr/core/array.py:2957]:
```suggestion
        new_attributes : dict
```

**Review by jhamman — CHANGES_REQUESTED** [review]
Thanks for pushing on this @MSanKeys963. A few minor suggestions but it mostly looks good.

**Review by jhamman — APPROVED** [review]

**Pull request merged**